dali/ft/filecopy.cpp

```diff
@@ -412,10 +415,10 @@ bool FileTransferThread::performTransfer()
             OutputProgress newProgress;
             newProgress.deserializeCore(msg);
             newProgress.deserializeExtra(msg, 1);
+            newProgress.deserializeExtra(msg, 2);
```

Comment:
question: isn't this now unconditionally trying to deserialize the extra info (which won't be there if version 1) ?

https://github.com/hpcc-systems/HPCC-Platform/pull/15979#discussion_r855017663
conclusion from that thread in previous PR I thought was that must always check for remaining (>bytesRequired) to handle versions < 2.
If so, I'd expect this to be a single call with current version:
```
    newProgress.deserializeExtra(msg, currentProgressVersion);
```
and for deserializeExtra to unconditionally deserialize for v1, and check remaining>v2required to deserialize v2 meta, if above v2, then it knows it's forced compatibility by all version >=2 sending version down and forcing min compact. to be sent back.

Reply:
Yes, Gavin pointed out that unconditionally deserializing v2 would work fine.  The reason is that if the 'done' is read in (line 411) with a value true, it breaks out of the loop.  And that means that when 'ok' is read (line 428),  will not read the v2 message.  Basically, there is no messages after the previous v1 message, so adding v2 message after v1 should work fine.